x/crosschain/keeper/abci.go

```diff
@@ -52,46 +52,47 @@ func (k Keeper) IterateAndUpdateCctxGasPrice(
 
 IterateChains:
 	for _, chain := range chains {
-		// support only external evm chains
-		if zetachains.IsEVMChain(chain.ChainId, additionalChains) && !zetachains.IsZetaChain(chain.ChainId, additionalChains) {
-			res, err := k.ListPendingCctx(sdk.UnwrapSDKContext(ctx), &types.QueryListPendingCctxRequest{
-				ChainId: chain.ChainId,
-				Limit:   gasPriceIncreaseFlags.MaxPendingCctxs,
-			})
-			if err != nil {
-				ctx.Logger().Info("GasStabilityPool: fetching pending cctx failed",
-					"chainID", chain.ChainId,
-					"err", err.Error(),
-				)
-				continue IterateChains
-			}
+		if !IsCCTXGasPriceUpdateSupported(chain.ChainId, additionalChains) {
+			continue
+		}
+
+		res, err := k.ListPendingCctx(sdk.UnwrapSDKContext(ctx), &types.QueryListPendingCctxRequest{
+			ChainId: chain.ChainId,
+			Limit:   gasPriceIncreaseFlags.MaxPendingCctxs,
+		})
+		if err != nil {
+			ctx.Logger().Info("GasStabilityPool: fetching pending cctx failed",
+				"chainID", chain.ChainId,
+				"err", err.Error(),
+			)
+			continue IterateChains
+		}
 
-			// iterate through all pending cctx
-			for _, pendingCctx := range res.CrossChainTx {
-				if pendingCctx != nil {
-					gasPriceIncrease, additionalFees, err := updateFunc(ctx, k, *pendingCctx, gasPriceIncreaseFlags)
-					if err != nil {
-						ctx.Logger().Info("GasStabilityPool: updating gas price for pending cctx failed",
-							"cctxIndex", pendingCctx.Index,
+		// iterate through all pending cctx
+		for _, pendingCctx := range res.CrossChainTx {
+			if pendingCctx != nil {
```

Comment:
NIT : think it might cleaner to 
```suggestion
			if pendingCctx == nil {
			 continue
			}
```

Reply:
yeah https://github.com/zeta-chain/node/pull/3377/commits/279335bf423ac338cf96c984e6bb27df7e68eb61